src/basic/hashmap.c

```diff
@@ -1976,3 +1976,25 @@ IteratedCache* iterated_cache_free(IteratedCache *cache) {
 
         return mfree(cache);
 }
+
+bool set_equal(Set *a, Set *b) {
+        void *p;
+
+        /* Checks whether each entry of 'a' is also in 'b' and vice versa, i.e. the two sets contain the same
+         * entries */
+
+        if (set_size(a) != set_size(b)) /* Cheap check that hopefully catches a lot of inequality cases
+                                         * already */
+                return false;
+
+        SET_FOREACH(p, a)
+                if (!set_contains(b, p))
+                        return false;
+
+        SET_FOREACH(p, b)
```

Comment:
If both set size are equal, checking that every element of `a` is in `b` should be enough, right ?
This is not necessary to check that every element of `b` is in `a`.

Reply:
You have a point. But only if the hash functions/equality functions for both sets are the same. I posted an updated PR now in #18667 now, where I made such a tweak. Thanks for reporting back